node/node.go

```diff
@@ -589,6 +593,60 @@ func (n *Node) SignBlobs(blobs []*core.BlobMessage, referenceBlockNumber uint) (
 	return signatures, nil
 }
 
+// ValidateBlobHeadersRoot validates the blob headers root hash
+// by comparing it with the merkle tree root hash of the blob headers.
+// It also checks if all blob headers have the same reference block number
+func (n *Node) ValidateBlobHeadersRoot(ctx context.Context, blobHeaderHashes [][32]byte, root [32]byte) error {
```

Comment:
Naming seems a little misleading. It's doing the following:
1. Check that all blobs have been previously validated and stored. 
2. Check reference block numbers
3. Check root

`ValidateBatchContents`?